pkg/virt-handler/vm.go

```diff
@@ -551,22 +560,25 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 								// mounted, and still in spec, and in phase we can change, update status to mounted.
 								volumeStatus.Phase = v1.HotplugVolumeMounted
 								volumeStatus.Message = fmt.Sprintf("Volume %s has been mounted in virt-launcher pod", volumeStatus.Name)
-								volumeStatus.Reason = "VolumeMountedToPod"
+								volumeStatus.Reason = VolumeMountedToPodReason
+								d.recorder.Event(vmi, k8sv1.EventTypeNormal, volumeStatus.Reason, volumeStatus.Message)
 							}
 						} else {
 							// Not mounted, check if the volume is in the spec, if not update status
 							if _, ok := specVolumeMap[volumeStatus.Name]; !ok && canUpdateToUnmounted(volumeStatus.Phase) {
 								// Not mounted.
 								volumeStatus.Phase = v1.HotplugVolumeUnMounted
 								volumeStatus.Message = fmt.Sprintf("Volume %s has been unmounted from virt-launcher pod", volumeStatus.Name)
-								volumeStatus.Reason = "VolumeUnMountedFromPod"
+								volumeStatus.Reason = VolumeUnMountedFromPodReason
+								d.recorder.Event(vmi, k8sv1.EventTypeNormal, volumeStatus.Reason, volumeStatus.Message)
 							}
 						}
 					} else {
 						// Successfully attached to VM.
 						volumeStatus.Phase = v1.VolumeReady
 						volumeStatus.Message = fmt.Sprintf("Successfully attach hotplugged volume %s to VM", volumeStatus.Name)
-						volumeStatus.Reason = "VolumeReady"
+						volumeStatus.Reason = VolumeReadyReason
+						d.recorder.Event(vmi, k8sv1.EventTypeNormal, volumeStatus.Reason, volumeStatus.Message)
```

Comment:
I just realized that we're rebuilding the volume status here on each reconcile loop.  Similar to my other comment about ensuring we only fire an event once (when possible), we're going to want to do that here as well. These events should only fire if something changes, not if the same result is calculated again.

Reply:
You make a very good point. Let me fix that.